### DISCUSSION TITLE
[POLL] Paid Offline Local Editor?

### DISCUSSION THREAD

**Discussion by davidleejy** [typst/typst#2163] [Polls]
Personally I think it would be sweet if Typst could release a local offline editor 😍📝. 

Moreover, with the open-sourcing of Typst's compiler, the doors to building local editors are open to developers and Typst enthusiasts.

**Comment by Enivex** [comment]
There are plans for a PWA version of the web app. Both will eventually be monetized.

**Comment by KillTheMule** [comment]
For me it's missing another option: I wouldn't care for an extra editor, no matter the licensing. I'd rather see typst-lsp developed some more (which, of course, isn't the question here, just saying).

**Comment by GroophyLifefor** [comment]
I used to do it, but I stopped because of the heavy legal rules of typst, it was easy to start, difficult to advance. I suggest you continue from web-editor

**Comment by laurmaedje** [comment]
We _might_ also ship the app as a desktop app at some point. It would still be based on web technologies, but would have local file system support etc.

**Comment by dei-layborer** [comment]
I'm all for the typst project finding ways to support itself financially, but I'm thinking this may be a hard sell.  Of course, I say this with the caveat that I can only guess at what typst's actual market actually is.  But that guess is that anyone who cares about editing offline will just use their own text editor + the CLI, while anyone who really wants the more "full" experience or however you want to describe it will use the web editor.

I for one use ``micro``, and this actually reminds me that I need to create a syntax highlighting definition for it to work with .typ files.  I've dabbled with the web editor and definitely like it, it just doesn't do anything for me specifically most of the time.  I will say that it is much better when I'm trying to figure out something new thanks to the IDE-like suggestions.  I also really appreciate that it allows you to upload outside files (fonts, images, etc.).  But even then, typst is competing with Office365 and Overleaf, both of which are much more thoroughly entrenched.

My personal thought (again, with little to nothing in terms of actual support) is that, until typst is more popular, paywalls of whatever sort are just going to impede adoption.  Most people (and businesses) use Word, while the scientific world uses TeX.  To be clear, I prefer typst to either, even in this early state, although of course there are bound to be people whose use-case exceeds what typst can do for the moment.
